### ISSUE TITLE
Multiple Billing Entities Stops Creation of Organization Accounts

### ISSUE THREAD

**Issue by mswanson-mastery** [Snowflake-Labs/terraform-provider-snowflake#2392]
### Terraform CLI and Provider Versions

```
Terraform v1.3.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azurerm v3.7.0
+ provider registry.terraform.io/hashicorp/local v2.4.1
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/snowflake-labs/snowflake v0.67.0

Your version of Terraform is out of date! The latest version
is 1.7.0. You can update by downloading from https://www.terraform.io/downloads.html
```

### Use Cases or Problem Statement

Last week we had our Snowflake contract enhanced to allow us to create accounts billed to either Azure or AWS. When running a manual `terraform apply` in our account provisioning project today to set up a new account on Azure, I was met with this error: 

```
╷
│ Error: 042120 (22023): Multiple suitable billing entities exist for the target cloud 'AZURE', please specify the desired billing entity.
│ 
│   with snowflake_account.managed_snowflake_accounts["CLIENT-NAME"],
│   on main.tf line 13, in resource "snowflake_account" "managed_snowflake_accounts":
│   13: resource "snowflake_account" "managed_snowflake_accounts" {
│ 
╵
```

I don't see any way to get around this, other than a manual account create described [here](https://community.snowflake.com/s/article/ERROR-Multiple-suitable-billing-entities-exist-for-the-target-cloud) passing in a value for `consumption_billing_entity`. 

`consumption_billing_entity` does not appear to be an available flag in any of the versions of the Snowflake provider that I've looked at, but I believe that it should be included as an optional flag for `snowflake_account`.

### Proposal

Add optional flag for `consumption_billing_entity` to allow for more configuration.

### How much impact is this issue causing?

Medium

**Comment by sfc-gh-asawicki** [comment]
Hey @mswanson-mastery. Thanks for reporting the issue.

We have supporting missing parameters on our roadmap https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#19012024-roadmap-overview. I will bump its priority. If it becomes a blocking issue, please use the [unsafe execute resource](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/unsafe_execute). It has some limitations, so please the docs carefully before using it.

**Comment by mswanson-mastery** [comment]
Hey there, @sfc-gh-asawicki - thank you for the workaround suggestion! It looks like that's not available on my version of hte Snowflake provider (0.67) so any bump in priority would be much appreciated. I am going to attempt to manually create the account in our primary instance of Snowflake and then import it into our State.

Here's the error for reference: 
```
╷
│ Error: Invalid resource type
│ 
│   on main.tf line 57, in resource "snowflake_unsafe_execute" "create_snowflake_account":
│   57: resource "snowflake_unsafe_execute" "create_snowflake_account" {
│ 
│ The provider snowflake-labs/snowflake does not support resource type "snowflake_unsafe_execute".
```

**Comment by sfc-gh-asawicki** [comment]
Hey @mswanson-mastery.

Yes, the unsafe execute resource is available from [v0.77.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.77.0).

I want to remind you that we are in the 0.x.x version. When we introduce the missing parameter you will still have to upgrade to the newest released version since we are not providing changes for older versions (check the [disclaimer in our README](https://github.com/Snowflake-Labs/terraform-provider-snowflake/tree/v0.77.0?tab=readme-ov-file#snowflake-terraform-provider) for more info).

**Comment by mswanson-mastery** [comment]
@sfc-gh-asawicki - That makes a lot of sense to me, we'll keep an eye out for that feature and plan to upgrade when it becomes available. 

For anyone else encountering this issue: Manually creating the account via a CREATE in Snowflake did allow for a seamless import into our existing state, so we do have a workaround that is functional and isn't too much of a lift-and-shift from our standard processes.

**Comment by chrisluedtke** [comment]
I see [consumption_billing_entity_name](https://registry.terraform.io/providers/snowflakedb/snowflake/latest/docs/resources/account#consumption_billing_entity_name-1) is available after creation. Is there an updated timeline for input support?

**Comment by sfc-gh-asawicki** [comment]
Yes, it will be available in 2.2.0 (https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#new-feature-new-consumption_billing_entity-field-in-snowflake_account-resource) which we plan to release on Monday/Tuesday.

**Comment by sfc-gh-jcieslak** [comment]
Hey @mswanson-mastery as Artur mentioned the consumption_billing_entity is now supported in the account resource. It should be specified if your organization contains more than one billing entity. Please, bump the provider version and use this option.
